sunpy/map/mapbase.py

```diff
@@ -1963,50 +1963,122 @@ def draw_grid(self, axes=None, grid_spacing: u.deg = 15*u.deg, annotate=True, **
 
-    def draw_limb(self, axes=None, **kwargs):
+    def draw_limb(self, axes=None, *, resolution=1000, **kwargs):
         """
-        Draws a circle representing the solar limb
+        Draws the solar limb as seen by the map's observer.
+
+        The limb is a circle for only the simplest plots.  If the coordinate frame of
+        the limb is different from the coordinate frame of the plot axes, not only
+        may the limb not be a true circle, a portion of the limb may be hidden from
+        the observer.  In that case, the circle is divided into visible and hidden
+        segments, represented by solid and dotted lines, respectively.
 
         Parameters
         ----------
-        axes: `~matplotlib.axes` or None
-            Axes to plot limb on or None to use current axes.
+        axes : `~matplotlib.axes` or ``None``
+            Axes to plot limb on or ``None`` to use current axes.
+        resolution : `int`
+            The number of points to use to represent the limb.
 
         Returns
         -------
-        circ: list
-            A list containing the `~matplotlib.patches.Circle` object that
-            has been added to the axes.
+        visible : `~matplotlib.patches.Polygon`
+            The patch added to the axes for the visible part of the limb (i.e., the
+            "near" side of the Sun).
+        hidden : `~matplotlib.patches.Polygon`
+            The patch added to the axes for the hidden part of the limb (i.e., the
+            "far" side of the Sun).
 
         Notes
         -----
-        Keyword arguments are passed onto `matplotlib.patches.Circle`.
+        Keyword arguments are passed onto the patches.
+
+        If the limb is a true circle, ``visible`` will instead be
+        `~matplotlib.patches.Circle` and ``hidden`` will be ``None``.
+
+        To avoid trigger Matplotlib auto-scaling, these patches are added as generic
+        artists instead of patches.  One consequence is that the plot legend is not
+        populated automatically when the limb is specified with a text label.
         """
         # Put import here to reduce sunpy.map import time
         from matplotlib import patches
+        from matplotlib.path import Path
 
         # Don't use _check_axes() here, as drawing the limb works fine on none-WCSAxes,
         # even if the image is rotated relative to the axes
         if not axes:
             axes = wcsaxes_compat.gca_wcs(self.wcs)
+        is_wcsaxes = wcsaxes_compat.is_wcsaxes(axes)
 
-        transform = wcsaxes_compat.get_world_transform(axes)
-        if wcsaxes_compat.is_wcsaxes(axes):
-            radius = self.rsun_obs.to(u.deg).value
-        else:
-            radius = self.rsun_obs.value
-        c_kw = {'radius': radius,
-                'fill': False,
+        c_kw = {'fill': False,
                 'color': 'white',
-                'zorder': 100,
-                'transform': transform
-                }
+                'zorder': 100}
```

Comment:
Whilst the artist/patch is being changed, I would recommend setting this to 1, which is default for `Patches`: https://matplotlib.org/stable/gallery/misc/zorder_demo.html. This should be adequate to show above a `WCSAxes`, which is treated by Matplotlib as a single `Artist`:

https://github.com/astropy/astropy/blob/5f8866e41164293e463a3cc53fadb5566ba488cc/astropy/visualization/wcsaxes/core.py#L30

for which the default zorder is 0:

https://github.com/matplotlib/matplotlib/blob/ba373b8862d9d458a118db04b6922f67e1115912/lib/matplotlib/artist.py#L97

Reply:
I don't object to this change, but I'm going to kick this to a future PR.  It breaks a handful of figure tests (particularly with `CompositeMap`), and I don't want to deal with that all in this PR.